### PR TITLE
CDD-2092: Wired in offline retry into online unhappy path.

### PR DESCRIPTION
**Pull request by googley42**

### CODE REVIEW

File: test/acceptance/CustomsNotificationRetryFailureSpec.scala [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+    stopMockServer()
+    await(repo.drop)
+  }
+
+  feature("Ensure offline retry") {
+    scenario("backend submits a valid PUSH request with incorrect callback details used resulting in endpoint returning a 404") {
+
+      startApiSubscriptionFieldsService(validFieldsId,callbackData)
+      Given("the Push endpoint is setup to return NOT_FOUND")
+      setupPushNotificationServiceToReturn(NOT_FOUND)
```
Comment (paulblakehmrc) [inline test/acceptance/CustomsNotificationRetryFailureSpec.scala]: test passes when this line is removed, like below. Is there something wrong with these acceptance tests?

File: test/unit/services/CustomsNotificationMetricsServiceSpec.scala [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+      Eventually.eventually(verify(mockMetricsConnector, times(1)).post(metricsRequestCaptor.capture()))
+      val metricsRequest = metricsRequestCaptor.getValue
+      metricsRequest.conversationId.toString shouldBe conversationId.id.toString
+      ()
+    }
+  }
+
+  "CustomsNotificationMetricsService" should {
+    "send notification metric when NotificationWorkItem has a metrics time" in new SetUp {
+      when(mockMetricsConnector.post(metricsRequest)).thenReturn(Future.successful())
```
[the remark this answers is not in the document]
Comment (googley42) [reply test/unit/services/CustomsNotificationMetricsServiceSpec.scala]: will do

File: test/acceptance/CustomsNotificationRetryFailureSpec.scala (hunk 1 above)
Comment (googley42) [reply test/acceptance/CustomsNotificationRetryFailureSpec.scala]: `setupPushNotificationServiceToReturn(NOT_FOUND)` will program a failure response. Not having this line will have the same effect as this means the endpoint does not respond successfully.

File: test/acceptance/CustomsNotificationRetryFailureSpec.scala [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+
+      And("pull queue was not called")
+      verifyNotificationQueueServiceWasNotCalled()
+    }
+
+    scenario("backend submits a valid PULL request but pull queue is unavailable") {
+
+      startApiSubscriptionFieldsService(validFieldsId, DeclarantCallbackDataOneForPull)
+      Given("the Push endpoint is setup to return NOT_FOUND")
+      runNotificationQueueService(NOT_FOUND)
```
[the remark this answers is not in the document]
Comment (googley42) [reply test/acceptance/CustomsNotificationRetryFailureSpec.scala]: `runNotificationQueueService(NOT_FOUND)` will program a failure response. Not having this line will have the same effect as this means the endpoint does not respond successfully.

**Pull request merged**